lib/test/EffectsTest.cpp

```diff
@@ -1,25 +1,57 @@
+#include <array>
 #include <cstdio>
 #include <interface/InterfaceController.hpp>
 #include <iostream>
 #include <map>
 #include <string>
-#include <array>
 
 #include "DummyParamController.hpp"
 #include "Effect.hpp"
 #include "Pole.hpp"
 #include "gtest/gtest.h"
 
-constexpr auto effect_names = std::array{"HuePoles", "BackAndForth",
-                                               "Sliders", "SideToSide"};
+constexpr auto effect_names =
+    std::array{"HuePoles", "BackAndForth", "Sliders", "SideToSide"};
 // Effect index starts at this value.
 constexpr uint8_t kEffectOffset = 3;
 
+uint16_t Brightness(const CRGB& rgb) { return rgb.r + rgb.g + rgb.b; }
+
 class EffectsTest : public ::testing::Test {
-  protected:
+ protected:
   Poles poles{};
   DummyParamController param_controller;
   InterfaceController controller{poles, std::addressof(param_controller)};
+
+  static uint8_t GetLightCount(const Pole& pole) {
+    uint8_t light_count = 0;
+    for (auto column : pole.get_grid_lights()) {
+      for (CRGB rgb : column) {
+        if (Brightness(rgb) > 0) {
+          light_count++;
+        }
+      }
+    }
+    return light_count;
+  }
+
+  uint16_t GetTotalLightCount() {
+    uint16_t light_count = 0;
+    for (Pole& pole : poles) {
+      light_count += GetLightCount(pole);
+    }
+    return light_count;
+  }
+
+  uint8_t GetPolesOn() {
+    uint8_t poles_on = 0;
+    for (Pole& pole : poles) {
+      if (GetLightCount(pole) > 0) {
+        poles_on++;
+      }
+    }
+    return poles_on;
```

Comment:
This entire body can be replaced with `return std::count_if(poles.begin(), poles.end(), [](Pole const & pole) { return GetLightCount(pole) > 0; });`

Reply:
Done. Nifty.